File: frontend/src/api/roomType.ts
import { request } from './request'

export interface RoomTypeVO {
  id: number
  name: string
  description?: string
  rackRate: number
  bedType?: string
  windowType?: string
  nonSmoking?: number
  maxGuests?: number
  status?: number
}

export interface RoomTypeForm {
  name: string
  description?: string
  rackRate: number
  bedType?: string
  windowType?: string
  nonSmoking?: number
  maxGuests?: number
  status?: number
}

export function listRoomTypesApi() {
  return request.get<{ data: RoomTypeVO[] }>('/room-types')
}

export function createRoomTypeApi(data: RoomTypeForm) {
  return request.post<{ data: RoomTypeVO }>('/room-types', data)
}

export function updateRoomTypeApi(id: number, data: RoomTypeForm) {
  return request.put<{ data: RoomTypeVO }>(`/room-types/${id}`, data)
}

File: backend/src/main/java/com/hotel/grms/module/room/entity/RoomType.java
package com.hotel.grms.module.room.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 房型实体，对应门市价与可售房型配置。
 *
 * @author liuxinsi
 * @date 2026-05-21
 */
@TableName("room_type")
public class RoomType {

    @TableId(type = IdType.AUTO)
    private Long id;
    private String name;
    private String description;
    private BigDecimal rackRate;
    private String bedType;
    private String windowType;
    private Integer nonSmoking;
    private Integer maxGuests;
    private Integer status;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public BigDecimal getRackRate() {
        return rackRate;
    }

    public void setRackRate(BigDecimal rackRate) {
        this.rackRate = rackRate;
    }

    public String getBedType() {
        return bedType;
    }

    public void setBedType(String bedType) {
        this.bedType = bedType;
    }

    public String getWindowType() {
        return windowType;
    }

    public void setWindowType(String windowType) {
        this.windowType = windowType;
    }

    public Integer getNonSmoking() {
        return nonSmoking;
    }

    public void setNonSmoking(Integer nonSmoking) {
        this.nonSmoking = nonSmoking;
    }

    public Integer getMaxGuests() {
        return maxGuests;
    }

    public void setMaxGuests(Integer maxGuests) {
        this.maxGuests = maxGuests;
    }

    public Integer getStatus() {
        return status;
    }

    public void setStatus(Integer status) {
        this.status = status;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }
}
